tests/src/test/java/org/eclipse/hono/tests/DeviceRegistryHttpClient.java

```diff
@@ -532,6 +542,45 @@ private static String registrationWithoutIdUri(final String tenant) {
         return httpClient.delete(requestUri, status -> status == HttpURLConnection.HTTP_NO_CONTENT);
     }
 
+    /**
+     * Finds devices belonging to the given tenant with optional filters, paging and sorting options.
+     *
+     * @param tenantId The tenant that the device belongs to.
+     * @param pageSize The maximum number of results to include in a response.
+     * @param pageOffset The offset into the result set from which to include objects in the response.
+     * @param filters The filters are predicates that objects in the result set must match.
+     * @param sortOptions A list of sort options.
+     * @param expectedStatus The status code indicating a successful outcome.
+     * @return A future indicating the outcome of the operation. The future will contain the response payload
+     *         if the request succeeded. Otherwise the future will fail with a 
+     *         {@link org.eclipse.hono.client.ServiceInvocationException}.
+     * @throws NullPointerException if the tenant is {@code null}.
+     */
+    public Future<Buffer> searchDevices(
+            final String tenantId,
+            final Optional<Integer> pageSize,
+            final Optional<Integer> pageOffset,
+            final Optional<List<String>> filters,
+            final Optional<List<String>> sortOptions,
```

Comment:
can we use the simple parameter types? I do not see why we need to use Optionals here. There is certainly no problem with providing a page size and an offset and empty lists, if no filters nor sort options are used, right?

Reply:
The page size and offset are not mandatory fields. I thought its better to test some cases without page size or offset. Hence I defined them as Optional parameters.

Reply:
So, we remove the Optionals for the Lists but keep them for the Integer params?

Reply:
I have removed the Optionals for the Lists and also rebased the code for PR#2412. Would you mind taking a look again?